kairos-prover/kairos-circuit-logic/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use alloc::{boxed::Box, string::String, vec::Vec};
+
+use account_trie::{Account, AccountTrie};
+use kairos_trie::{stored::merkle::Snapshot, DigestHasher, NodeHash, TrieRoot};
+use sha2::Sha256;
+use transactions::{KairosTransaction, L1Deposit, Signed, Withdraw};
+
+pub mod account_trie;
+pub mod transactions;
+
+/// `ProofInputs` contains the minimum logical inputs needed to apply a batch of transactions.
+///
+/// The trie snapshot holds the pre-batch merkle root.
+/// The post-batch merkle root is given running `self.run_batch_proof_logic`
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ProofInputs {
+    pub transactions: Box<[KairosTransaction]>,
+    pub trie_snapshot: Snapshot<Account>,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, Eq, PartialEq, PartialOrd, Ord)]
+pub struct ProofOutputs {
+    pub pre_batch_trie_root: TrieRoot<NodeHash>,
+    pub post_batch_trie_root: TrieRoot<NodeHash>,
+    /// TODO consider replacing with a count and hash of the processed deposits
+    pub deposits: Box<[L1Deposit]>,
```

Comment:
I think `L1Deposit` should have additional `event_id` field, to uniquely identify deposit. Then proof outputs could contain list of deposit IDs, or even a single number: maximum deposit event ID.

Reply:
We could just make this a Vec of Strings and use hex::encode(deposit.hash())

Reply:
This is something different.
By hashing here I mean hashing over all the deposits in the batch.
That gives a single hash to serialize, but comes at the cost of an additional sha256 in and outside the vm.

Not sure if it's worth it, but it's a possible future optimization.